runtime/near-vm-runner/src/logic/recorded_storage_counter.rs

```diff
@@ -0,0 +1,38 @@
+use super::errors::InconsistentStateError;
+use super::{HostError, VMLogicError};
+
+/// Keeps track of the recorded storage proof size and ensures it does not exceed the limit.
+pub struct RecordedStorageCounter {
+    initial_storage_size: usize,
+    last_observed_storage_size: usize,
+    size_limit: usize,
+}
+
+impl RecordedStorageCounter {
```

Comment:
I'm wondering if we can push this implementation to the ext.rs layer? We could have this check done for every trie operations like `storage_set`, `storage_get`, `storage_remove`, instead of remembering to have to call it in logic/mod.rs

Reply:
The way I see it per-receipt storage counting is very similar to what we have with gas, so it would be nice to have those in the same place to keep it somewhat consistent. Current implementation does exactly that, so I suggest to keep it.